src/main/java/org/skriptlang/skript/bukkit/recipes/RecipeUtils.java

```diff
@@ -0,0 +1,215 @@
+		CRAFTING(getCraftingRecipeClass(), RegisterRecipeEvent.CraftingRecipeEvent.class),
+		BLASTING(BlastingRecipe.class, RegisterRecipeEvent.CookingRecipeEvent.BlastingRecipeEvent.class),
+		FURNACE(FurnaceRecipe.class, RegisterRecipeEvent.CookingRecipeEvent.FurnaceRecipeEvent.class),
+		CAMPFIRE(CampfireRecipe.class, RegisterRecipeEvent.CookingRecipeEvent.CampfireRecipeEvent.class),
+		SMOKING(SmokingRecipe.class, RegisterRecipeEvent.CookingRecipeEvent.SmokingRecipeEvent.class),
+		COOKING(CookingRecipe.class, RegisterRecipeEvent.CookingRecipeEvent.class), // Having 'COOKING' under the subclasses allows for proper ExprRecipeType
+		SMITHING_TRANSFORM(SmithingTransformRecipe.class, RegisterRecipeEvent.SmithingRecipeEvent.SmithingTransformRecipeEvent.class),
+		SMITHING_TRIM(SmithingTrimRecipe.class, RegisterRecipeEvent.SmithingRecipeEvent.SmithingTrimRecipeEvent.class),
+		SMITHING(SmithingRecipe.class, RegisterRecipeEvent.SmithingRecipeEvent.class), // Having 'SMITHING' under the subclasses allows for proper ExprRecipeType
+		STONECUTTING(StonecuttingRecipe.class, RegisterRecipeEvent.StonecuttingRecipeEvent.class),
+		COMPLEX(ComplexRecipe.class, null);
+
+		private final @Nullable Class<? extends Recipe> recipeClass;
+		private final @Nullable Class<? extends Event> eventClass;
+
+		RecipeType(@Nullable Class<? extends Recipe> recipeClass, @Nullable Class<? extends Event> eventClass) {
+			this.recipeClass = recipeClass;
+			this.eventClass = eventClass;
+		}
+
+		/**
+		 * Gets the Bukkit recipe type class.
+		 * @return Bukkit recipe class
+		 */
+		public @Nullable Class<? extends Recipe> getRecipeClass() {
+			return recipeClass;
+		}
+
+		/**
+		 * Gets the custom event used when creating a new recipe.
+		 * @return Custom event class
+		 */
+		public @Nullable Class<? extends Event> getEventClass() {
+			return eventClass;
+		}
+
+		// Due to 1.19 not having 'CraftingRecipe.class'
+		private static @Nullable Class<? extends Recipe> getCraftingRecipeClass() {
+			if (Skript.classExists("org.bukkit.inventory.CraftingRecipe"))
+				return CraftingRecipe.class;
+			return null;
+		}
+	}
+
+	/**
+	 * Gets {@link RecipeType} from provided recipe class.
+	 * @param providedClass Bukkit recipe class
+	 * @return Recipe Type
+	 */
+	public static RecipeType getRecipeTypeFromRecipeClass(Class<? extends Recipe> providedClass) {
+		for (RecipeType type : RecipeType.values()) {
+			if (type.recipeClass != null && type.recipeClass.isAssignableFrom(providedClass)) {
+				return type;
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * Gets {@link RecipeType} from provided recipe.
+	 * @param providedRecipe Recipe
+	 * @return Recipe Type
+	 */
+	public static RecipeType getRecipeTypeFromRecipe(Recipe providedRecipe) {
+		return getRecipeTypeFromRecipeClass(providedRecipe.getClass());
+	}
+
+	// Custom Events used for SecRegisterRecipe
+	public static class RegisterRecipeEvent extends Event {
```

Comment:
i think this should be in its own file
it'd also make the whole long-name situation up at the top of utils less egregious